native/lib/evm.go

```diff
@@ -217,20 +189,24 @@ func (s *Service) EvmApply(params EvmParams) (error, *EvmResult) {
 		// - The EVM.create call can fail before it even reaches the point of incrementing the nonce. We have to make
 		//   sure to NOT decrement the nonce in that case. Hence, setting the nonce to the value before the EVM call in
 		//   case it was modified.
-		nonce := statedb.GetNonce(params.From)
+		nonce := statedb.GetNonce(invocation.Caller)
 		if nonce > 0 {
-			statedb.SetNonce(params.From, nonce-1)
+			statedb.SetNonce(invocation.Caller, nonce-1)
```

Comment:
True. This logic only works for top-level calls. For nested calls (originating form a native SC) this would lead to a missing nonce increment. None of the native SC deploy contract currently, but there is nothing preventing them from doing so. So this is absolutely a valid bug.